### PR TITLE
send http.route attribute in flask metric

### PR DESCRIPTION
**Pull request by GonzaloGuasch**
# Description

Taking up [this closed PR](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2106), when sending metrics in the Flask instrumentation, the http.route attribute is added to them.
This attribute is super important in the use of metrics and will greatly improve the experience of using Flask. This change was tested locally.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
 running: tox -e  test-instrumentation-flask

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [ ] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Comment by xrmx** [comment]
@GonzaloGuasch please sign the CLA

**Comment by xrmx** [comment]
@GonzaloGuasch please do all your commits from the same email that signed the CLA, see https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2506#issuecomment-2099119631

**Comment by GonzaloGuasch** [comment]
Changing the PR to draft until i resolve CLA certification with mi company!

File: CHANGELOG.md [hunk 1]
```diff
@@ -47,6 +47,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- `opentelemetry-instrumentation-flask` Add http route to metric attributes
+  ([#2506](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2506))
+-
```
Comment (xrmx) [inline CHANGELOG.md:52]:
```suggestion
```

File: instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py [hunk 2]
```diff
@@ -388,13 +392,21 @@ def _start_response(status, response_headers, *args, **kwargs):
...
             duration_histogram_old.record(
                 max(round(duration_s * 1000), 0), duration_attrs_old
             )
         if duration_histogram_new:
             duration_attrs_new = otel_wsgi._parse_duration_attrs(
                 attributes, _HTTPStabilityMode.HTTP
             )
+
+            if wrapped_app_environ.get(_ENVIRON_REQUEST_ROUTE_KEY, None):
+                duration_attrs_new[SpanAttributes.HTTP_ROUTE] = wrapped_app_environ.get(_ENVIRON_REQUEST_ROUTE_KEY)
```
Comment (emdneto) [inline instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py:408]: Shouldn't we use the new semconv attribute here?

**Review by emdneto — CHANGES_REQUESTED** [review]
Please fix lint

File: instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py (hunk 2 above)
Comment (GonzaloGuasch) [reply instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py]: What do you mean by new semconv? To change the variable?
It is mandatory to have the key value "http.route"

**Comment by xrmx** [comment]
@GonzaloGuasch the CLA check is failing :sweat:

**Pull request closed without merging**